src/Propel/Generator/Builder/Om/ObjectBuilder.php

```diff
@@ -894,6 +894,8 @@ public function addTemporalAccessorComment(&$script, Column $column)
      * @return string|{$dateTimeClass}{$orNull} Formatted date/time value as string or $dateTimeClass object (if format is NULL), NULL if column is NULL" . ($handleMysqlDate ? ', and 0 if column value is ' . $mysqlInvalidDateString : '') . "
      *
      * @throws PropelException - if unable to parse/validate the date/time value.
+     *
+     * @psalm-return (\$format is null ? {$dateTimeClass}{$orNull} : string{$orNull})
```

Comment:
The linked docs says this only works with template parameters, but [here](https://psalm.dev/articles/conditional-love) it says this is a short hand syntax that works, too, so I guess it is ok

Reply:
Hey @mringler, somewhere else you wrote:

>When updating my project, I realized that in your other psalm PR, where you added `@psalm-return ($format is null ? DateTime : string)` to DateTime getters, I did not realize that these function also return `null`. You might want to fix that.
Correct type is probably
`@psalm-return null|($format is null ? DateTime : string)`

As you see, the returned type is `{$dateTimeClass}{$orNull}` or `string{$orNull}`, so if the field you're referencing is nullable, the `null` typehint should already be there!

Reply:
Oh, you are right. The columns I saw were all required. Should have checked again. Sorry!